sql/src/main/java/io/crate/action/job/ContextPreparer.java

```diff
@@ -412,6 +412,24 @@ public ExecutionSubContext visitCollectPhase(final CollectPhase phase, final Pre
         }
 
         @Override
+        public ExecutionSubContext visitFileUriCollectPhase(FileUriCollectPhase phase, PreparerContext context) {
```

Comment:
FileUriCollectPhase doesn't have a nodePageSizeHint which is used to get the RowReceiver in visitRoutedCollectPhase. The other stuff is just calling a ctor so I didn't factor it into a separate method. But I can change that if you want.